proxy/api/src/main.rs

```diff
@@ -83,15 +83,51 @@ async fn main() -> Result<(), Box<dyn std::error::Error>> {
     let bin_dir = config::bin_dir()?;
     coco::git_helper::setup(&proxy_path, &bin_dir).expect("Git remote helper setup failed");
 
-    log::info!("Starting API");
     let ctx = context::Ctx::from(context::Context {
         peer_api,
         keystore,
         store,
     });
+
+    {
+        let ctx = ctx.clone();
+        log::info!("Starting Announcement watcher");
+        tokio::task::spawn(async move { announcement_watcher(ctx).await });
+    }
+
+    log::info!("Starting API");
     let api = http::api(ctx, args.test);
 
     warp::serve(api).run(([127, 0, 0, 1], 8080)).await;
 
     Ok(())
 }
+
+async fn announcement_watcher(ctx: context::Ctx) {
+    // TODO(xla): Take interval timings from config/settings.
+    let mut timer = tokio::time::interval(std::time::Duration::from_secs(10));
+
+    loop {
```

Comment:
Why is it safe to `unwrap` inside the loop? If one of these panics, the spawned future will die (probably silently).

Reply:
Indeed, not quite sure how to address this without channels and selects over the different tasks we spawn in main.

Reply:
Can't you catch and log the errors and `continue` in the `loop`?